kubernetes/kubernetes.go

```diff
@@ -240,12 +241,24 @@ func newKubeAgent(ctx context.Context, config KubeAgentConfig) KubeAgentConfig {
 	return config
 }
 
+//nolint gocyclo
 func (ka KubeAgentConfig) collectMetrics(ctx context.Context, config KubeAgentConfig,
 	clientset kubernetes.Interface, nodeSource NodeSource) (rerr error) {
 
 	sampleStartTime := time.Now().UTC()
 
-	//create metric sample directory
+	// refresh client token before each collection
+	token, err := getBearerToken(config.BearerTokenPath)
+	if err != nil {
+		log.Warnf("Couldn't read auth token defined in %q: %v", config.BearerTokenPath, err)
```

Comment:
After discussing with the team we decided we do just want to warn here because the metrics-agent on older versions will continue to successfully poll metrics and for newer clusters, this will warn a couple of times and then their metrics-agent will start receiving 400s of stale token requests. This will give users a good description on why their metrics-agent is not working